internal/provider/resources/work_queue.go

```diff
@@ -0,0 +1,367 @@
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Work queue ID (UUID)",
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was created (RFC3339)",
+			},
+			"account_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Account ID (UUID), defaults to the account set in the provider",
+				Optional:    true,
+			},
+			"workspace_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Workspace ID (UUID), defaults to the workspace set in the provider. In Prefect Cloud, either the `work_pool` resource or the provider's `workspace_id` must be set.",
+				Optional:    true,
+			},
+			"updated": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was updated (RFC3339)",
+			},
+			"work_pool_name": schema.StringAttribute{
+				Description: "The name of the work pool associated with this work queue",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the work queue",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "Description of the work queue",
+				Default:     stringdefault.StaticString(""), // Because prefect returns this as the default for none provided
+			},
+			"is_paused": schema.BoolAttribute{
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+				Description: "Whether this work queue is paused",
+				Optional:    true,
+			},
+			"concurrency_limit": schema.Int64Attribute{
+				Description: "The concurrency limit applied to this work queue",
+				Optional:    true,
+			},
+			"priority": schema.Int64Attribute{
+				Description: "The priority of this work queue",
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+// copyWorkQueueToModel maps an API response to a model that is saved in Terraform state.
+func copyWorkQueueToModel(queue *api.WorkQueue, tfModel *WorkQueueResourceModel) {
+	tfModel.ID = types.StringValue(queue.ID.String())
+	tfModel.Created = customtypes.NewTimestampPointerValue(queue.Created)
+	tfModel.Updated = customtypes.NewTimestampPointerValue(queue.Updated)
+	tfModel.ConcurrencyLimit = types.Int64PointerValue(queue.ConcurrencyLimit)
+	tfModel.Priority = types.Int64PointerValue(queue.Priority)
+	tfModel.Description = types.StringPointerValue(queue.Description)
+	tfModel.Name = types.StringValue(queue.Name)
+	tfModel.IsPaused = types.BoolValue(queue.IsPaused)
+	tfModel.WorkPoolName = types.StringValue(queue.WorkPoolName)
+}
+
+func StringPtr(s string) *string {
+	return &s
+}
```

Comment:
I don't mind having this since it's so small, but you can use `ptr.To` instead of having to define your own here ([example](https://github.com/PrefectHQ/terraform-provider-prefect/blob/90ce565433daccc51e6ba51b7ae3cc393c332e8c/internal/api/automations_test.go#L36)).